src/Collection/CollectionTrait.php

```diff
@@ -135,10 +135,20 @@ public function map(callable $c)
      * {@inheritDoc}
      *
      */
-    public function reduce(callable $c, $zero)
+    public function reduce(callable $c, $zero = null)
     {
+        $skip = false;
+        if ($zero === null) {
+            $zero = $this->first();
+            $skip = true;
+        }
+
         $result = $zero;
         foreach ($this->_unwrap() as $k => $value) {
+            if ($skip) {
+                $skip = false;
+                continue;
```

Comment:
I think it is a fait tradeoff. If you don't want to pass a `$zero` it is normal to expect that the first value will not be passed to the callback.